src/lib/tls/tortls.c

```diff
@@ -1856,407 +379,31 @@ tor_tls_free_(tor_tls_t *tls)
     size_t r,w;
     tor_tls_get_n_raw_bytes(tls,&r,&w); /* ensure written_by_tls is updated */
   }
-#ifdef SSL_set_tlsext_host_name
-  SSL_set_tlsext_host_name(tls->ssl, NULL);
-#endif
-  SSL_free(tls->ssl);
+  tor_tls_impl_free_(tls->ssl);
```

Comment:
Why not have a `FREE_AND_NULL()` variant of this?

Reply:
tls_impl_free_ and tls_x509_impl_free_ are tiny wrappers around the underlying types.